cub/cub/device/dispatch/dispatch_select_if.cuh

```diff
@@ -62,6 +62,22 @@
 
 CUB_NAMESPACE_BEGIN
 
+namespace detail
+{
+/**
+ * @brief Wrapper that partially specializes the `AgentSelectIf` on the non-type name parameter `KeepRejects`.
+ */
+template <bool KeepRejects>
```

Comment:
I am not sure whether this is feasible in this PR, but I always bullied @gevtushenko to move towards enum classes instead of plain booleans for things like this

Reply:
I am totally fine with punting on this